cedar-policy-cli/src/lib.rs

```diff
@@ -688,30 +680,51 @@ fn read_policy_and_links(
     policies_filename: impl AsRef<Path>,
     links_filename: Option<impl AsRef<Path>>,
 ) -> Result<PolicySet> {
-    let mut pset = read_policy_file(policies_filename.as_ref())?;
+    let mut pset = read_policy_set(Some(policies_filename.as_ref()))?;
     if let Some(links_filename) = links_filename {
         add_template_links_to_set(links_filename.as_ref(), &mut pset)?;
     }
     Ok(pset)
 }
 
-fn read_policy_file(filename: impl AsRef<Path>) -> Result<PolicySet> {
-    let src = std::fs::read_to_string(filename.as_ref()).context(format!(
-        "failed to open policy file {}",
-        filename.as_ref().display()
-    ))?;
-    let ps = PolicySet::from_str(&src).context(format!(
-        "failed to parse policies from file {}",
-        filename.as_ref().display()
+fn read_from_file_or_stdin(filename: Option<impl AsRef<Path>>, context: &str) -> Result<String> {
```

Comment:
Can we add some docs so users can hover over `read_from_file_or_stdin` and `read_from_file` and see why one would be used vs the other? Personally I find it confusing that `read_from_file` calls `read_from_file_or_stdin`